turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -1165,6 +1167,106 @@ impl ModuleGraph {
         }
         Ok(())
     }
+
+    /// Traverses all reachable nodes and also continue revisiting them as long the visitor returns
+    /// GraphTraversalAction::Continue. The visitor is responsible for the runtime complexity and
+    /// eventual termination of the traversal. This corresponds to computing a fixed point state for
+    /// the graph.
+    ///
+    /// Nodes are (re)visited according to the returned priority of the node, prioritizing high
+    /// values.
+    ///
+    /// * `entries` - The entry modules to start the traversal from
+    /// * `state` - The state to be passed to the callbacks
+    /// * `visit` - Called for a specific edge
+    ///    - Receives: (originating &SingleModuleGraphNode, edge &ChunkingType), target
+    ///      &SingleModuleGraphNode, state &S
+    ///    - Return [GraphTraversalAction]s to control the traversal
+    /// * `priority` - Called for before visiting the children of a node to determine its priority.
+    ///    - Receives: target &SingleModuleGraphNode, state &S
+    ///    - Return a priority value for the node
+    ///
+    /// Returns the number of node visits (i.e. higher that the node count if there are revisits).
+    pub async fn traverse_edges_fixed_point<S, P: Ord>(
+        &self,
+        entries: impl IntoIterator<Item = (ResolvedVc<Box<dyn Module>>, P)>,
+        state: &mut S,
+        mut visit: impl FnMut(
+            Option<(&'_ SingleModuleGraphModuleNode, &'_ ChunkingType)>,
+            &'_ SingleModuleGraphModuleNode,
+            &mut S,
+        ) -> GraphTraversalAction,
+        priority: impl Fn(&'_ SingleModuleGraphModuleNode, &mut S) -> P,
+    ) -> Result<usize> {
+        let graphs = self.get_graphs().await?;
+
+        #[derive(PartialEq, Eq)]
+        struct NodeWithPriority<T: Ord> {
```

Comment:
nit: Is there a benefit of making this generic and defined inline in the function?

Reply:
What do you mean? It is already generic over `T` and define inside the function body?

Reply:
I meant that we are instantiating it only with a single type, so I wanted to check if conversion to a generic type is intentional.

Reply:
Ah, yes. For scope hoisting I am going to reuse the traversal function but use a different priority.